srml/contracts/rpc/src/lib.rs

```diff
@@ -121,4 +138,57 @@ where
 
 		Ok(exec_result)
 	}
+
+	fn get_storage(
+		&self,
+		address: AccountId,
+		key: Bytes,
+		at: Option<<Block as BlockT>::Hash>,
+	) -> Result<Option<Bytes>> {
+		let api = self.client.runtime_api();
+		let at = BlockId::hash(at.unwrap_or_else(||
+			// If the block hash is not supplied assume the best block.
+			self.client.info().best_hash));
+
+		if key.len() != 32 {
+			return Err(Error {
+				code: ErrorCode::InvalidParams,
+				message: format!(
+					"The key is expected to be 32 bytes long and only {} bytes are given",
+					key.len(),
+				),
+				data: None,
+			});
+		}
+
+		let mut key_fixed_len = [0u8; 32];
+		key_fixed_len.copy_from_slice(&*key);
+
+		let get_storage_result = api
+			.get_storage(&at, address, key_fixed_len)
+			.map_err(|e|
+				// Handle general API calling errors.
+				Error {
+					code: ErrorCode::ServerError(RUNTIME_ERROR),
+					message: "Runtime trapped while querying storage.".into(),
+					data: Some(format!("{:?}", e).into()),
+				})?
+			.map_err(|e|
+				// Convert contract errors to RPC errors.
+				match e {
+					GetStorageError::ContractDoesntExist => Error {
```

Comment:
Maybe a `From<GetStorageError> for Error`?

Reply:
This code is in crate A. `GetStorageError` is declared in crate `B`. `Error` is declared in crate `C`. I think orphan rules won't allow to do this? Unless you have something in mind that I don't : )

Reply:
Right, I guess we could wrap in a `struct GetStorageError(other::GetStorageError)` and then just `map_err(GetStorageError)?`.

Having a separate function would be ok too. I just feel bad seeing the "meat" of this function being diluted with tedious error handling.

Reply:
Ok, you were right, this looks better!